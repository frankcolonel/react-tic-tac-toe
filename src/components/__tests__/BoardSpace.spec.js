import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import BoardSpace from '../BoardSpace';

function setupComponent() {
  const props = {
    column: 2,
    index: 1,
    value: 1,
    handleNewMark: jest.fn(),
    isWinnerMark: jest.fn(),
    getPlayerSymbol: () => 'X',
  };

  const wrapper = shallow(
    <BoardSpace {...props} />,
  );

  return {
    props,
    wrapper,
  };
}

describe('<BoardSpace />', () => {
  it('should render without exploding', () => {
    const { wrapper } = setupComponent();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle new mark on board space click', () => {
    const { wrapper, props } = setupComponent();

    wrapper.find('li').first().simulate('click');

    expect(props.handleNewMark.mock.calls).toEqual([
      [props.column, props.index],
    ]);
  });
});
