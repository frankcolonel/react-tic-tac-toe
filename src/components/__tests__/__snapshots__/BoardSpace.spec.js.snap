exports[`<BoardSpace /> should render without exploding 1`] = `
<li
  data-css-hjczo8=""
  onClick={[Function]}>
  <Icon
    name="times" />
</li>
`;
